docs/site/migration/models/remoting-hooks.md

```diff
@@ -6,8 +6,266 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/migration-models-remoting-hooks.html
 ---
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3950](https://github.com/strongloop/loopback-next/issues/3950).
-" %}
+## Introduction
+
+In LoopBack 3, a [Remote hook](https://loopback.io/doc/en/lb3/Remote-hooks.html)
+enables you to execute a function before or after a remote method is called by a
+client:
+
+- `beforeRemote()` runs before the remote method.
+- `afterRemote()` runs after the remote method has finished successfully.
+- `afterRemoteError()` runs after the remote method has finished with an error.
+
+Hooks can be configured to run for one or more methods using a wildcard
+specifier. In general, there are three kinds of hook scopes:
+
+- **Global hooks** are executed for _every_ remote method.
+
+  These hooks are registered on the application object using a `**` wildcard.
+  For example:
+
+  ```js
+  app.beforeRemote('**', handlerFn);
+  ```
+
+- **Model-level hooks** are executed for every remote method of a given model
+  class.
+
+  These hooks are typically registered on the application object using
+  `{model-name}.**` wildcard. For example, to run a hook for all methods of the
+  `User` model:
+
+  ```js
+  app.beforeRemote('User.**', handlerFn);
+  ```
+
+- **Method-level hooks** are executed only for a single method of a given model
+  class.
+
+  These hooks are typically registered on the model class. For example, to run a
+  hook whenever `User.login` is called remotely:
+
+  ```js
+  User.beforeRemote('User.login', handlerFn);
+  ```
+
+LoopBack 4 provides [Interceptors](../../Interceptors.md) feature to enable
+application developers to implement similar functionality. In the following
+sections, we will first explain how to rewrite hook implementations to
+interceptors and then show how to register interceptors to be invoked globally,
+at a class level or at a method level.
+
+## Migrating a hook to an interceptor
+
+Let's take a look at a typical interceptor implementation:
```

Comment:
I think this part and `### Example` are vague in which part belongs to LB3 or LB4.

Maybe: `Let's take a look at a typical interceptor implementation in LB4`: